recipes/daw_json_link/all/conanfile.py

```diff
@@ -12,70 +17,81 @@ class DawJsonLinkConan(ConanFile):
     homepage = "https://github.com/beached/daw_json_link"
     topics = ("json", "parse", "json-parser", "serialization", "constexpr", "header-only")
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
     no_copy_source = True
     short_paths = True
 
-    _compiler_required_cpp17 = {
-        "Visual Studio": "16",
-        "gcc": "8",
-        "clang": "7",
-        "apple-clang": "12.0",
-    }
+    @property
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+        }
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        if tools.Version(self.version) < "2.11.0":
-            self.requires("daw_header_libraries/1.29.7")
-        elif tools.Version(self.version) < "2.12.0":
-            self.requires("daw_header_libraries/2.5.3")
-        else:
-            self.requires("daw_header_libraries/2.68.1")
+        self.requires("daw_header_libraries/2.72.0")
         self.requires("daw_utf_range/2.2.2")
 
-    def validate(self):
-        if self.settings.get_safe("compiler.cppstd"):
-            tools.check_min_cppstd(self, "17")
+    def package_id(self):
+        self.info.clear()
+
+    @property
+    def _info(self):
+        return self if Version(conan_version).major < 2 else self.info
```

Comment:
I think we're supposed to avoid using `conan_version` now: https://github.com/conan-io/conan-center-index/pull/13703#discussion_r1002827863

Reply:
As you expect, I use `conan_version` by referencing header_only templates.
I'm waiting an answer of your question in #13703.

Reply:
For self.info and header-only I don't see another alternative. Conan team still don't have an answer how to deal with it. I asked about it here: https://github.com/conan-io/conan/issues/12210 and continued here: https://github.com/conan-io/docs/pull/2778

The `env_info` on the other hand, is already prepared to be fixed on the next Conan 2 beta release.

One thing is correct to affirm, we will need to update it for all recipes in the future. Or because we need to use only `self.settings` with validate() because Conan 1.x requires, or because we have this new method which uses or not self.info

For now, we can use `def _info(self)` as we don't have a resolution for this situation.